### PR TITLE
feat: implement CSS Coverage

### PR DESCRIPTION
**Pull request by aslushnikov**
This patch adds two new methods to the `page.coverage` namespace:
- `page.coverage.startCSSCoverage()` - to initiate css coverage
- `page.coverage.stopCSSCoverage()` - to stop css coverage

The coverage format is consistent with the JavaScript coverage.

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -2170,22 +2214,6 @@ Identifies what kind of target this is. Can be `"page"`, `"service_worker"`, or
 > **NOTE** JavaScript Coverage doesn't include anonymous scripts; however, scripts with sourceURLs are
 reported.
```
Comment (JoelEinbinder) [inline docs/api.md:2218]: What happens if there is an injected inline style?

File: test/test.js [hunk 2]
```diff
@@ -3489,6 +3489,70 @@ describe('Page', function() {
...
+        expect(coverage.length).toBe(2);
+      });
+      it('should NOT report scripts across navigations', async function({page, server}) {
+        await page.coverage.startCSSCoverage(); // Enabled by default.
+        await page.goto(server.PREFIX + '/csscoverage/multiple.html');
+        await page.goto(server.EMPTY_PAGE);
+        const coverage = await page.coverage.stopCSSCoverage();
+        expect(coverage.length).toBe(0);
+      });
+    });
```
Comment (JoelEinbinder) [inline test/test.js:3569]: golden file please

File: lib/Coverage.js [hunk 3]
```diff
@@ -127,6 +139,104 @@ class JSCoverage {
...
+   */
+  async start(options = {}) {
+    console.assert(!this._enabled, 'CSSCoverage is already enabled');
+    this._resetOnNavigation = options.resetOnNavigation === undefined ? true : !!options.resetOnNavigation;
+    this._enabled = true;
+    this._stylesheetURLs.clear();
+    this._stylesheetSources.clear();
+    this._eventListeners = [
+      helper.addEventListener(this._client, 'CSS.styleSheetAdded', this._onStyleSheet.bind(this)),
+      helper.addEventListener(this._client, 'Runtime.executionContextsCleared', this._onExecutionContextsCleared.bind(this)),
```
Comment (JoelEinbinder) [inline lib/Coverage.js:179]: It’s a bit odd to listen for executionContextCleared for stylesheets. Might this race against something?

File: lib/Coverage.js [hunk 4]
```diff
@@ -127,6 +139,104 @@ class JSCoverage {
...
+      this._stylesheetURLs.set(header.styleSheetId, header.sourceURL);
+      this._stylesheetSources.set(header.styleSheetId, response.text);
+    } catch (e) {
+      // This might happen if the page has already navigated away.
+      debugError(e);
+    }
+  }
+
+  /**
+   * @return {!Promise<!Array<!{url:string, text:string, ranges:!Array<!{start:number, end:number}>}>>}
```
Comment (JoelEinbinder) [inline lib/Coverage.js]: not: can we have this typedef as CoverageData

Comment (JoelEinbinder) [reply lib/Coverage.js]: nit* To force the same format

**Review by JoelEinbinder — APPROVED** [review]

File: test/test.js (hunk 2 above)
Comment (aslushnikov) [reply test/test.js:3569]: Done.

File: lib/Coverage.js (hunk 3 above)
Comment (aslushnikov) [reply lib/Coverage.js:179]: They're all sent from the same process, so no races here

File: lib/Coverage.js (hunk 4 above)
Comment (aslushnikov) [reply lib/Coverage.js]: Done

File: docs/api.md (hunk 1 above)
Comment (aslushnikov) [reply docs/api.md:2218]: The style will be ignored similarly to the anonymous scripts in js coverage. Added a test and a doc note for this.

**Comment by paulirish** [comment]
There's a few open issues regarding coverage. Perhaps some of them can be closed, thanks to this PR? That should raise the visibility and get some early users.

**Pull request merged**